client/src/redux/index.js

```diff
@@ -180,21 +180,26 @@ export const reducer = handleActions(
       ...state,
       userProfileFetchState: { ...defaultFetchState }
     }),
-    [actionTypes.fetchUserComplete]: (state, { payload: { user } }) => ({
-      ...state,
-      user: {
-        ...state.user,
-        sessionUser: user
-      },
-      currentChallengeId:
-        user?.currentChallengeId || store.get(CURRENT_CHALLENGE_KEY),
-      userFetchState: {
-        pending: false,
-        complete: true,
-        errored: false,
-        error: null
-      }
-    }),
+    [actionTypes.fetchUserComplete]: (state, { payload: { user } }) => {
+      return {
+        ...state,
+        user: {
+          ...state.user,
+          sessionUser: {
+            ...state.user.sessionUser,
+            ...user
+          }
```

Comment:
What was the goal here? As it stands a signed out user will get an empty user object, so appear to be signed in. But not completely, so anything (like Growthbook) that tries to make use of the user object will not work.

Reply:
Every time this ran, the `sessionUser` was overwritten. This PR needs the `sessionUser.examEnvironmentAuthorizationToken` to persist in the state, even if `/get-session-user` is refetched.

I did not think anything would change here, because we went from:

```diff
- user: { sessionUser: newSessionUser }
+ user: { sessionUser: { ...previousSessionUser, ...newSessionUser }
```

Reply:
I could move the token out of `sessionUser`, but thought that weird because the `userToken` is already in there - I wanted to keep them together.